src/strategy_manager/strategy.manager.cpp

```diff
@@ -0,0 +1,157 @@
+/******************************************************************************
+ * Copyright © 2013-2019 The Komodo Platform Developers.                      *
+ *                                                                            *
+ * See the AUTHORS, DEVELOPER-AGREEMENT and LICENSE files at                  *
+ * the top-level directory of this distribution for the individual copyright  *
+ * holder information and the developer policies on copyright and licensing.  *
+ *                                                                            *
+ * Unless otherwise agreed in a custom licensing agreement, no part of the    *
+ * Komodo Platform software, including this file may be copied, modified,     *
+ * propagated or distributed except according to the terms contained in the   *
+ * LICENSE file                                                               *
+ *                                                                            *
+ * Removal or modification of this copyright notice is prohibited.            *
+ *                                                                            *
+ ******************************************************************************/
+
+#include <vector>
+#include <unordered_map>
+
+#include "strategy.manager.hpp"
+
+namespace antara {
+  bool asset::operator==(const asset &other) const {
+    return symbol == other.symbol;
+  };
+
+  std::pair<st_quote, st_base> pair::to_std_pair() {
+    return std::pair<st_quote, st_base>(quote, base);
+  };
+
+  bool pair::operator==(const pair &rhs) const {
+    return base == rhs.base && quote == rhs.quote;
+  };
+}
+
+namespace std {
+
+  template <>
+  struct hash<antara::pair> {
+    std::size_t operator()(const antara::pair& p) const {
+      using std::size_t;
+      using std::hash;
+
+      std::size_t h1 = std::hash<std::string>{}(p.base.value());
+      std::size_t h2 = std::hash<std::string>{}(p.quote.value());
+
+      return h1 ^ (h2 << 1);
+    }
+  };
+}
+
+enum class Side { BUY, SELL, BOTH };
+
+class market_making_strategy {
```

Comment:
Same here, if it's contains only data and no constructor it's should be a struct